typescript/agentkit/src/wallet-providers/evmWalletProvider.ts

```diff
@@ -2,14 +2,60 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
 import { WalletProvider } from "./walletProvider";
-import { TransactionRequest, ReadContractParameters, ReadContractReturnType } from "viem";
+import { TransactionRequest, ReadContractParameters, ReadContractReturnType, PublicClient, EstimateGasReturnType, EstimateGasParameters, Chain, EstimateFeesPerGasParameters, FeeValuesEIP1559 } from "viem";
+
+/**
+ * Configuration for gas limit and gas fee multipliers.
+ */
+export interface EvmWalletProviderGasConfig {
+  /**
+   * A internal multiplier on gas limit estimation.
+   */
+  gasLimitMultiplier?: number;
+
+  /**
+   * A internal multiplier on fee per gas estimation.
+   */
+  feePerGasMultiplier?: number;
+}
+
+/**
+ * Configuration options for the EVM Providers.
+ */
+export interface EvmWalletProviderConfig {
+  /**
+   * A RPC client.
+   */
+  publicClient: PublicClient;
```

Comment:
Lets not make the assumption that all `EvmWalletProvider` will have a viem `PublicClient`. I would prefer to keep the gas estimation method up to the implementor. For example `CdpWalletProvider` will eventually do gas estimation in our internal backend vs going direct to nodes which will remove the need for a `PublicClient`